File: examples/js/intercept.mjs
/**
 * Conditionally intercept requests and responses
 */
export const REQ_FILTER = `(path-matches r"^/login$")`
export const RES_FILTER = REQ_FILTER

export function onRequest(req) {
    log.info("Intercepting login request")
    return req.intercept()
}

export function onResponse(res) {
    log.info("Intercepting login response")
    return res.intercept()
}

File: examples/python/intercept.py
"""
Conditionally intercept requests and responses
"""

RES_FILTER = REQ_FILTER = """
(path-matches r"^/login$")
"""

def on_request(req):
    log.info("Intercepting login request")
    return req.intercept()


def on_response(res):
    log.info("Intercepting login response")
    return res.intercept()
